lib/foreman/renderer.rb

```diff
@@ -345,7 +345,7 @@ def preseed_attributes
     end
 
     def yast_attributes
-      @dynamic   = @host.diskLayout =~ /^#Dynamic/ if (@host.respond_to?(:disk) && @host.disk.present?) || @host.ptable.present?
+      @dynamic   = !!(@host.disk_layout_template[:template] =~ /^#Dynamic/) if @host.disk_layout_template
```

Comment:
Is the change here from `.diskLayout` to `.disk_layout_template` intentional?

Reply:
I based this on https://github.com/theforeman/foreman/pull/6120/files#diff-6632a0077c474a44423343eb597f9d72R333 since that fixed an infinite loop, so without the change from `.diskLayout` to `.disk_layout_template` the related/added tests would fail.

Reply:
It's probably best to get #6120 in first, we can then rebase this PR on top of #6120 and everything will be nice and clean.